andela/mervels-bn-backend
# #168131432 Add Search Functionality

#### What does this PR do?
add request search functionality
#### Description of Task to be completed?
add ability for manager to search requests by
- `id` - which is the requestId
- `requester` - which can pass first name or last name of the requester
- `status` - which can be `pending, approved, rejected`
- `travelDate or returnDate` - requests with format `yyyy-mm-dd`
- `from` - represents the origin
- `reason` - any word in reason 
- `accommodation` - search by accommodation name
- `destination` - search by destination name

#### How should this be manually tested?
- clone the repo and checkout to `ft-search-requests-168131432` branch
- run `npm install` then `npm run test` to check it tests pass
- Login as a manager and head to postman on `api/v1/search/requests` endpoint  with GET method to get all requests.
- Alternatively you can search by any params as shown in the description 
`api/v1/search/requests?status=pending&requester=Jonathan`
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[ #168131432](https://www.pivotaltracker.com/story/show/168131432)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/18899285/65668236-a3603c80-e041-11e9-9069-5b92d417d370.png)
![image](https://user-images.githubusercontent.com/18899285/65668356-db677f80-e041-11e9-82b0-f340b2ba79e8.png)
![image](https://user-images.githubusercontent.com/18899285/65743637-c51afb80-e0f5-11e9-97d8-4c825a2aa842.png)


#### Questions:
N/A
